modules/mono/SCsub

```diff
@@ -33,11 +34,15 @@ if env.editor_build:
     env_mono.add_source_files(env.modules_sources, "editor/*.cpp")
     SConscript("editor/script_templates/SCsub")
 
-    if env["build_assemblies"] or env["generate_mono_glue"]:
-        gd = os.path.join(env.Dir("#bin").abspath, "godot" + env["PROGSUFFIX"])
+    if env_local["build_csharp"]:
+       env_local["build_assemblies"] = True
+       env_local["generate_mono_glue"] = True
```

Comment:
I'm trying to understand why the `env_local` clone is needed here.

The only difference I notice is this, so I assume you did it this way to prevent tampering with `env["build_assemblies"]` and `env["generate_mono_glue"]` which might be set by the user? Since we override them anyway, the values passed by users don't matter much so this doesn't seem needed to me.

What _could_ be done however to improve things a bit is to treat `build_csharp` as a convenience alias (like we do with `production`, `dev_build` and `dev_mode` in the main `SConstruct`, that changes default values but still allows overriding). This would be done like this:

```python
env["build_assemblies"] = methods.get_cmdline_bool("build_assemblies", env["build_csharp"])
env["generate_mono_glue"] = methods.get_cmdline_bool("generate_mono_glue", env["build_csharp"])
```

This means that the value for e.g. `build_assemblies` will be in priority what the user specified on the command line, or the value of `build_csharp` otherwise.

So `scons build_csharp=yes` would enable both `build_assemblies` and `generate_mono_glue`, but you can still opt-out of either with `scons build_csharp=yes build_assemblies=no` - the user specified value takes precedence. The latter is of course equivalent to `scons build_csharp=no generate_mono_glue=yes`, it's just another way to do it (and might be useful if we eventually make `build_csharp=yes` the default, so you can easily opt out of either step).

Reply:
Ah, yeah, that was what I was trying to do, not trample all over user settings, but didn't know how to, thanks!